compiler/ir/backend.jvm/codegen/src/org/jetbrains/kotlin/backend/jvm/intrinsics/Not.kt

```diff
@@ -40,6 +39,15 @@ object Not : IntrinsicMethod() {
             markLineNumber(expression)
             value.discard()
         }
+
+        override fun materializeAt(target: Type, irTarget: IrType, castForReified: Boolean) {
+            value.materializeAt(Type.BOOLEAN_TYPE, codegen.context.irBuiltIns.booleanType, false)
+            mv.iconst(1)
+            mv.xor(Type.INT_TYPE)
+            if (Type.BOOLEAN_TYPE != target) {
```

Comment:
When is it possible?

Reply:
Here is an example: 
https://github.com/JetBrains/kotlin/blob/7d59c7689c1c0fa18978a04a8d7bb3f505361867/compiler/testData/codegen/box/callableReference/function/booleanNotIntrinsic.kt

Reply:
OK, thanks. By the way, this condition should not be necessary, since `StackValue.coerce` will not do anything in case source and target types are equal.

Reply:
Makes sense, updated.